src/applications/detection/main.cpp

```diff
@@ -51,236 +52,353 @@ using boost::timer;
+  {
+    if(counter == 0)
+    {
+      CCTAG_COUT_NOENDL(marker.id() + 1);
     }
-    
-    std::cout << std::endl << nMarkers << " markers detected and identified" << std::endl;
+    else
+    {
+      CCTAG_COUT_NOENDL(", " << marker.id() + 1);
+    }
+    ++counter;
+  }
+
+  std::cout << std::endl << nMarkers << " markers detected and identified" << std::endl;
 }
 
 /*************************************************************/
 /*                    Main entry                             */
+
 /*************************************************************/
 int main(int argc, char** argv)
 {
   CmdLine cmdline;
 
-  if( cmdline.parse( argc, argv ) == false ) {
-    cmdline.usage( argv[0] );
+  if(cmdline.parse(argc, argv) == false)
+  {
+    cmdline.usage(argv[0]);
     return EXIT_FAILURE;
   }
 
-  cmdline.print( argv[0] );
+  cmdline.print(argv[0]);
+  
+  bool useCamera = false;
 
   // Check input path
-  if( cmdline._filename.compare("") != 0){
-    if (!boost::filesystem::exists( cmdline._filename )) {
+  if(!cmdline._filename.empty())
+  {
+    if(isInteger(cmdline._filename))
+    {
+      useCamera = true;
+    }
+    else if(!boost::filesystem::exists(cmdline._filename))
+    {
       std::cerr << std::endl
-        << "The input file \""<< cmdline._filename << "\" is missing" << std::endl;
+              << "The input file \"" << cmdline._filename << "\" is missing" << std::endl;
       return EXIT_FAILURE;
     }
-  }else{
+  }
+  else
+  {
     std::cerr << std::endl
-        << "An input file is required" << std::endl;
-    cmdline.usage( argv[0] );
+            << "An input file is required" << std::endl;
+    cmdline.usage(argv[0]);
     return EXIT_FAILURE;
   }
 
 #ifdef WITH_CUDA
-  popart::pop_cuda_only_sync_calls( cmdline._switchSync );
+  popart::pop_cuda_only_sync_calls(cmdline._switchSync);
 #endif
 
   // Check the (optional) parameters path
-  std::size_t nCrowns = std::atoi(cmdline._nCrowns.c_str());
+  const std::size_t nCrowns = std::atoi(cmdline._nCrowns.c_str());
   cctag::Parameters params(nCrowns);
-  
-  if( cmdline._paramsFilename != "" ) {
-    if (!boost::filesystem::exists( cmdline._paramsFilename )) {
+
+  if(!cmdline._paramsFilename.empty())
+  {
+    if(!boost::filesystem::exists(cmdline._paramsFilename))
+    {
       std::cerr << std::endl
-        << "The input file \""<< cmdline._paramsFilename << "\" is missing" << std::endl;
+              << "The input file \"" << cmdline._paramsFilename << "\" is missing" << std::endl;
       return EXIT_FAILURE;
     }
 
     // Read the parameter file provided by the user
-    std::ifstream ifs( cmdline._paramsFilename.c_str() );
+    std::ifstream ifs(cmdline._paramsFilename);
     boost::archive::xml_iarchive ia(ifs);
     ia >> boost::serialization::make_nvp("CCTagsParams", params);
     CCTAG_COUT(params._nCrowns);
     CCTAG_COUT(nCrowns);
-    assert( nCrowns == params._nCrowns );
-  } else {
+    assert(nCrowns == params._nCrowns);
```

Comment:
`assert` has no effect in release builds. Throw `std::runtime_error` or smth instead.